src/Extension.php

```diff
@@ -20,6 +20,9 @@ public function getConfigKey()
 
     public function initialize(ExtensionManager $extensionManager)
     {
+        if (PHP_MAJOR_VERSION === 5) {
+            trigger_error('Behatch context extension stop supporting PHP5 in version 4.0', E_USER_DEPRECATED);
```

Comment:
would write it like that:
`@trigger_error('PHP 5 support is deprecated since 3.x and will be removed in 4.0, E_USER_DEPRECATED);`

WDYT?

Reply:
> would write it like that:

Silencing the deprecation notices it’s great when you handle them with another component (for example, the symfony debug toolbar). But here there is nothing to handle notices, they will never displayed.

Reply:
but in this case, the code execution will be interrupted, or am I wrong?
And in this case, it is no soft deprecation...

But in the first place you want to tell the user, silently (via the logfile), that it IS still working, but WILL NOT be working with the next major version.
Otherwise you break PHP 5 support directly

We always use `@trigger_error` here: https://github.com/sonata-project

@greg0ire WDYT?

Reply:
http://symfony.com/doc/current/components/phpunit_bridge.html#trigger-deprecation-notices

http://symfony.com/doc/current/components/phpunit_bridge.html#write-assertions-about-deprecations

Reply:
> but in this case, the code execution will be interrupted, or am I wrong?

No, PHP5 tests still passed https://travis-ci.org/Behatch/contexts/jobs/356812969#L688 (with inelegant deprecation notice).

> http://symfony.com/doc/current/components/phpunit_bridge.html#trigger-deprecation-notices

How do you opt-in deprecation notices in behat?

Reply:
> Behatch context extension stop supporting PHP5 in version 4.0

"The behatch context extension will drop support for PHP 5 in version 4.0"

Regarding the `@`, it's fine with or without, it really is your choice. Putting the `@` makes it opt in, not putting it makes it opt out. Regarding behat, here is the error handling method: https://github.com/Behat/Behat/blob/efea9c2dea5694e8c029c392fe2bb7880925d9e1/src/Behat/Behat/Definition/Annotation/Definition.php#L119-L126
Not sure how to replace it with a better error handler.

Reply:
@greg0ire thank you for the corrections.

I found [caciobanu/behat-deprecation-extension](https://github.com/caciobanu/behat-deprecation-extension), but it may be a good idea to improve the behat error handler.

Reply:
In that case, I'd recommend adding the `@` and adding a link to the behat extension in the README

Reply:
Ok, I added a note in `UPGRADE-4.0.md` file.